build-tools/scripts/jenkinsfile

```diff
@@ -0,0 +1,484 @@
+            }
+
+            steps {
+                dir("$XADir") {
+                    sh "make all-tests CONFIGURATION=${env.BuildFlavor} V=1"
+                }
+
+                echo "Stage result: build tests: ${currentBuild.currentResult}"
+            }
+        }
+        
+        stage ('process build results') {
+            options {
+                timeout(time: 10, unit: 'MINUTES')      // Typically takes less than a minute
+            }
+
+            steps {
+                script {
+                    dir("$XADir") {
+                        def status = processBuildStatus()
+
+                        if (status == 0) {
+                            buildStatusProcessed = true
+                        }
+                    }
+                }
+
+                echo "Stage result: process build results: ${currentBuild.currentResult}"
+            }
+        }
+
+        stage('publish packages to Azure') {
+            options {
+                timeout(time: 10, unit: 'MINUTES')    // Typically takes less than 2 minutes
+            }
+
+            steps {
+                script {
+                    def status = publishPackages(publishBuildFilePaths)
+                    if (status == 0) {
+                        buildStatusUploaded = true
+                    }
+                }
+
+                echo "Stage result: publish packages to Azure: ${currentBuild.currentResult}"
+            }
+        }
+        
+        stage('run all tests') {
+            options {
+                timeout(time: 160, unit: 'MINUTES')     // XA tests typically take 1 hr, 40 minutes to run; timeout after 2 hrs, 40 minutes
+            }
+
+            steps {
+                dir("$XADir") {
+                    script {
+                        def status = 0
+                        try
+                        {
+                            echo "running tests"
+                            status = sh(
+                                script: "make run-all-tests CONFIGURATION=${env.BuildFlavor} V=1",
+                                returnStatus: true
+                            );
+                        } catch (error) {
+                            echo "ERROR : Unexpected error: ${error}"
+                            status = 1
+                        } finally {
+                            if (status != 0) {
+                                echo "ERROR : test run failed with status ${status}. Marking build as UNSTABLE"
+                                currentBuild.result = 'UNSTABLE'
+                            }
+                        }
+
+                        echo "Stage result: run all tests: status: ${status} / current build result: ${currentBuild.currentResult}"
+                    }
+                }
+            }   
+        }
+        
+        stage('publish test error logs to Azure') {
+            when {
+                expression { isPR == true }
```

Comment:
We should probably just skip this; we've had instances where this would have been useful for master builds as well.

Reply:
Fixed with commit https://github.com/xamarin/xamarin-android/commit/0fa360c82d7f426cc81697fcd7bce56412362024